frontend/pulse3d/src/components/account/AdminAccountOptions.js

```diff
@@ -0,0 +1,174 @@
+    if (updateStatus === UPDATE_STATUSES.SUCCESS) {
+      return <SaveButtonTextContainer style={{ color: "green" }}>Update Successful!</SaveButtonTextContainer>;
+    } else if (updateStatus === UPDATE_STATUSES.FAILURE) {
+      return (
+        <SaveButtonTextContainer style={{ color: "red" }}>Error, Update Failed.</SaveButtonTextContainer>
+      );
+    }
+    return <></>;
+  };
+
+  // get account settings at load
+  useEffect(() => {
+    getAccountSettings();
+  }, []);
+
+  const getAccountSettings = async () => {
+    const res = await fetch(`${process.env.NEXT_PUBLIC_USERS_URL}/${accountId}`);
+    const data = await res.json();
+    setAccountSettings(data);
+    setAccountSettingsEdits(data);
+  };
+
+  const updateErrorMessages = (newErrorMessages) => {
+    setErrorMessages({
+      ...errorMessages,
+      ...newErrorMessages,
+    });
+  };
+
+  const updateAccountSettingsEdits = (newValues) => {
+    setUpdateStatus(UPDATE_STATUSES.EDITING);
+
+    const { alias: newAlias } = newValues;
+    if (newAlias != null) {
+      let errorMsg = "";
+      if (newAlias.length > 0) {
+        if (newAlias.length < 6 || 128 < newAlias.length) {
+          errorMsg = "Must be 6-128 characters long";
+        } else if (/^\s/.test(newAlias) || /\s$/.test(newAlias)) {
+          errorMsg = "Cannot start or end with whitespace";
+        } else if (/\s\s/.test(newAlias)) {
+          errorMsg = "Cannot contain consecutive whitespace characters";
+        }
+      }
+      updateErrorMessages({ alias: errorMsg });
+      setAccountSettingsEdits({
+        ...accountSettingsEdits,
+        alias: newAlias,
+      });
+    }
+  };
+
+  const confirmChanges = async () => {
+    if (isButtonDisabled) return;
+
+    const requestBody = {
+      action_type: "set_alias",
+      new_alias: accountSettingsEdits.alias,
+    };
+
+    const newAccountSettings = {
+      ...accountSettings,
+      ...accountSettingsEdits,
+    };
+
+    setUpdateStatus(UPDATE_STATUSES.PENDING);
+
+    const res = await fetch(`${process.env.NEXT_PUBLIC_USERS_URL}/${accountId}`, {
+      method: "PUT",
+      body: JSON.stringify(requestBody),
+    });
+    if (res.status === 200) {
+      setAccountSettings(newAccountSettings);
+      setUpdateStatus(UPDATE_STATUSES.SUCCESS);
+    } else {
+      setUpdateStatus(UPDATE_STATUSES.FAILURE);
+    }
+  };
+
+  return (
+    <>
+      <InputContainer>
+        <FormInput
+          name="account_alias"
+          label="Account Alias"
+          placeholder={accountSettings.alias ? "Leave empty to remove the current alias" : "None set"}
+          value={accountSettingsEdits.alias || ""}
+          tooltipText={
+            "Set an alias for the Customer ID field used when logging into a user account (6-128 characters)."
+          }
+          onChangeFn={(e) => {
+            updateAccountSettingsEdits({ alias: e.target.value });
+          }}
+        />
+        <ErrorText id={"Account Alias Error"} role="errorMsg">
```

Comment:
Are parentheses needed in the ID?

Reply:
What do you mean by that?

Reply:
can't it be id="Account Alias Error" or id="accountAliasError"?

Reply:
yeah it probably could